### DISCUSSION TITLE
Read offline map cache storage from SDcard

### DISCUSSION THREAD

**Discussion by sonix-github** [strukturart/o.map#27] [Ideas]
Dear John-David,
thank you for great application. I am curious if there is some possibility to use offline maps from cache located on SDcard which are generated by **MOB**ile **A**tlas **C**reator (https://mobac.sourceforge.io). In my old Nokia C5 phone with Symbian I have used AFtrack application for hiking/cycling navigation completely offline. The maps for it I have created with MOBAC and then copied into phone to AFtrack cache folder.

Thanks

Best regards
sonix

**Comment by strukturart** [comment]
Mobac requires java, kaiOs does not support java. In  my app you can download with key * map parts for offline use. have you tried this ?

**Reply by sonix-github** [reply]
Dear John-David,

yes I tried download map parts via * key. Unfortunately I do not know how to "download" all map parts for all available zoom levels for area of my interest with radius around 60 km with this technique. My idea is to have downloaded all map parts for available zoom levels for area of my interest via MOBAC on desktop pc a then copy it to SDcard to some folder used as a map part cache and be able to read it from o.map. So not to create cache from o.map itself, but use already created cache with help of MOBAC which will be the same folder structure as used in your o.map

**Comment by strukturart** [comment]
Hello Sonix,

You can set the zoom depth of the cached map tiles in the settings: key 3 to open the menu.
The area of the map that is visible at the time you press key 3 is cached.

The plugin that takes care of the caching uses pouchDB, one possibility would be to export/import the content database. this would make it possible to exchange the data on different devices.

https://stackoverflow.com/questions/37229561/how-to-import-export-database-from-pouchdb

**Reply by sonix-github** [reply]
Dear John-David,

thanks for clarification about map tiles cashing and zoom depth. I will try it tomorrow. Is there some cache size limitation in the o.map application or kaios? Could you please clarify also option for offline maps "After how many days should the maps re-downloaded?" If I do not want to re-download should I leave it empty or set it to 0?

Thanks

Best regards
sonix

**Comment by strukturart** [comment]
> Could you please clarify also option for offline maps "After how many days should the maps re-downloaded?
if you do not enter a value, the map part will be tried to cache again after 24 hours

I use this plugin:
https://github.com/MazeMap/Leaflet.TileLayer.PouchDBCached

**Comment by strukturart** [comment]
@sonix-github are the extracted tiles that you created with MOBile Atlas Creator .png files? maybe there is a simple possibility to use it offline. can you provide me with the files from a small region?

**Reply by sonix-github** [reply]
Hello John-David,

yes, the map tiles are in .png format. I have attached mobac sample atlas for zoom levels 7 - 10.

Thanks for your interest.

Attachment for download: [MobacAtlasSample_zoom7-10.zip](https://github.com/strukturart/o.map/files/7138637/MobacAtlasSample_zoom7-10.zip)

**Comment by strukturart** [comment]
i have "asked around" what would be possible. i guess it is possible to create a folder structure within the app where the tiles would be stored. but this would mean that the app would be "bloated". i could make a version for you but you would have to sideload the app because it would not work for the normal user.

**Reply by sonix-github** [reply]
Hello John-David,

sideload is not problem for me. I do not understand what do you mean with "bloated". I thought that the caching can be optionally done to SDcard so the cache can be "replaced" for another one (downloaded/created via mobac) and copied to the SDcard from PC. So the o.map will be able to "read" map tiles cache from SDcard.

Best regards
sonix

**Comment by strukturart** [comment]
https://github.com/strukturart/o.map/blob/9ce9593d0c10b277638ae4fca0bd63216e69b674/application/assets/js/maps.js#L157

For security reasons, it is not possible to access files via file: // so the files must be located within the app or, as in my cache function, within a local database.

**Reply by sonix-github** [reply]
Thanks for clarification. And is there some possibility to read/write to that folder structure within the app from PC?

**Comment by strukturart** [comment]
https://strukturart.github.io/o.map/

you can also try out the app with a browser, but not everything works because the browser does not have the same api as kaios. but caching works.

**Reply by sonix-github** [reply]
Hmm, that's interesting :) Thanks for the tip. I am not familiar with web app development but I am going to learn about it to be able better understand how the app is working :) And thanks for your great o.map app.
